kubetools/deploy/image.py

```diff
@@ -49,19 +49,16 @@ def has_app_commit_image(registry, app_name, context_name, commit_hash):
 
 
 def _get_container_contexts_from_config(app_config):
-    context_name_to_build = {
-        key: context['build']
-        for key, context in app_config.get('containerContexts', {}).items()
-        if 'build' in context
-    }
-
+    context_name_to_build = {}
+    container_contexts = app_config.get('containerContexts', {})
     for deployment, data in app_config.get('deployments', {}).items():
         containers = data.get('containers')
         for name, container in containers.items():
             if 'containerContext' in container:
-                continue
+                context_name = container['containerContext']
+                context_name_to_build[context_name] = container_contexts[context_name]['build']
```

Comment:
Realising this doesn't account for the no build key case that was handled in the previous code. Will update